lib/local_daily_digest.js

```diff
@@ -30,18 +39,65 @@ function get(masterCb) {
 				lastDigestSent = new Date(lastDigestSent);
 				var offset = current.valueOf() - lastDigestSent.valueOf();
 				if (offset > 24 * 60 * 60 * 1000) { // If last digest was sent more than 24 hours ago
-	  				allLocalProfilesWDigest.push(profile);
-					// For all the profiles which want to get a digest, get a list of the locations they want a digest for
-					profile.dailyDigest.forEach(function (location) {
-						requiredLocationsId.push(location);
+					//if profile roles does not contain manager or editor
+
+	  				//forEach country
+	  				profile.dailyDigest.forEach(function (location) {
+		  				profile.roles.forEach(function(role){
+	  						if(role.split(/:(.+)?/)[1] != 'undefined'){
```

Comment:
I think this could/should be replaced by an equivalent of the hasRole function that you'll find in lib/roles.js: if (hasRole(profile, "manager:" + location) || hasRole(profile, "editor:" + location).

Ideally: add the hasRole function to the profile model, and then use profile.hasRole("manager:" + location). Makes sense ?